src/runtime-rs/crates/shimv2/src/shim_start.rs

```diff
@@ -0,0 +1,221 @@
+            .arg("-namespace")
+            .arg(&self.args.namespace)
+            .arg("-address")
+            .arg(&self.args.address)
+            .arg("-publish-binary")
+            .arg(&self.args.publish_binary)
+            .env("RUST_BACKTRACE", "1");
+
+        if self.args.debug {
+            command.arg("-debug");
+        }
+
+        Ok(command)
+    }
+
+    fn create_shim_process<T: IntoRawFd>(&self, socket: T) -> Result<u32> {
+        let mut cmd = self.new_command()?;
+        cmd.env(ENV_KATA_BIND_FD, format!("{}", socket.into_raw_fd()));
+        let child = cmd.spawn().map_err(Error::SpawnChild)?;
+
+        Ok(child.id())
+    }
+
+    fn get_shim_info_from_sandbox(&self, sandbox_id: &str) -> Result<(PathBuf, u32)> {
+        // All containers of a pod share the same pod socket address.
+        let address = self.socket_address(sandbox_id)?;
+        let bundle_path = self.get_bundle_path()?;
+        let sandbox_bundle_path = bundle_path
+            .join("..")
+            .join(sandbox_id)
+            .canonicalize()
+            .context(Error::GetBundlePath)?;
+        let pid = self.read_pid_file(&sandbox_bundle_path)?;
+
+        Ok((address, pid))
+    }
+}
+
+fn new_listener<T: AsRef<OsStr>>(address: T) -> Result<UnixListener> {
+    // Listen on an abstract Unix Domain Socket.
+    let mut bind_address = OsString::from("\0");
+    bind_address.push(address);
+    bind_address.push("\0");
+
+    UnixListener::bind(&bind_address).context("bind address")
+}
+
+#[cfg(test)]
+mod tests {
+    use serial_test::serial;
+    use tests_utils::gen_rand_string;
+
+    use super::*;
+    use crate::Args;
+
+    #[test]
+    #[serial]
+    fn test_new_command() {
+        let dir = tempfile::tempdir().unwrap();
+        let bundle_path = dir.path();
+        std::env::set_current_dir(bundle_path).unwrap();
+
+        let args = Args {
+            id: "sandbox1".into(),
+            namespace: "ns".into(),
+            address: "address".into(),
+            publish_binary: "containerd".into(),
+            socket: "socket".into(),
+            bundle: bundle_path.to_str().unwrap().into(),
+            debug: false,
+        };
+        let mut executor = ShimExecutor::new(args);
+
+        let cmd = executor.new_command().unwrap();
+        assert_eq!(cmd.get_args().len(), 8);
+        assert_eq!(cmd.get_envs().len(), 1);
+        assert_eq!(
+            cmd.get_current_dir().unwrap(),
+            executor.get_bundle_path().unwrap()
+        );
+
+        executor.args.debug = true;
+        let cmd = executor.new_command().unwrap();
+        assert_eq!(cmd.get_args().len(), 9);
+        assert_eq!(cmd.get_envs().len(), 1);
+        assert_eq!(
+            cmd.get_current_dir().unwrap(),
+            executor.get_bundle_path().unwrap()
+        );
+    }
+
+    #[test]
+    #[serial]
+    fn test_get_info_from_sandbox() {
+        let dir = tempfile::tempdir().unwrap();
+        let sandbox_id = gen_rand_string(16);
```

Comment:
This call is used a lot so I'd consider creating a test-specific `gen_cid()` or similar.

Reply:
Thanks for the review, change to `gen_id()` because it uses not only for cid